server/player/src/packet_handlers/placement.rs

```diff
@@ -24,66 +42,102 @@ pub fn handle_player_block_placement(
     packet_buffers
         .received::<PlayerBlockPlacement>()
         .for_each_valid(world, |world, (player, packet)| {
-            // TODO: handle slabs, blocks with directions, etc.
-            let gamemode = *world.get::<Gamemode>(player);
-            let inventory = world.get::<Inventory>(player);
-
-            let item = match inventory.item_in_main_hand(player, world) {
-                Some(item) => item,
-                None => return, // No block to place
-            };
-
-            drop(inventory);
-
-            let block = match item.ty.to_block() {
-                Some(block) => block,
-                None => return, // Item is not a block
-            };
-
-            let placed_on = match game.block_at(packet.location) {
+            let target_block = match game.block_at(packet.location) {
                 Some(block) => block,
                 None => {
-                    game.disconnect(player, world, "attempted to place block in unloaded chunk");
+                    game.disconnect(
+                        player,
+                        world,
+                        "Attempted to interact with block in unloaded chunk",
+                    );
                     return;
                 }
             };
 
-            // TODO: waterlogged blocks, more
-            let pos = match placed_on.kind() {
-                BlockKind::Grass | BlockKind::TallGrass | BlockKind::Water | BlockKind::Lava => {
-                    packet.location
-                }
-                _ => packet.location + packet.face.placement_offset(),
-            };
+            // Decide whether the player should place a block or interact with the block they are targeting
+            // TODO: Maybe player shifting may need to be taken into account (shift click on interactable block)
```

Comment:
This is something to do after sneaking gets implemented. Currently, the server isn't aware of it.

Reply:
Oh hahaha, thanks!